.github/scripts/periodic_merge.py

```diff
@@ -0,0 +1,146 @@
+from typing import Any, List
+
+from gitutils import GitRepo, get_git_remote_name, get_git_repo_dir
+from trymerge import (
+    GH_CHECKSUITES_FRAGMENT,
+    GH_COMMIT_AUTHORS_FRAGMENT,
+    GH_PR_REVIEWS_FRAGMENT,
+    GH_PULL_REQUEST_FRAGMENT,
+    GitHubPR,
+    MandatoryChecksMissingError,
+    check_for_sev,
+    find_matching_merge_rule,
+    gh_add_labels,
+    gh_graphql,
+    gh_remove_label,
+    handle_exception,
+    pr_get_failed_checks,
+    pr_get_pending_checks,
+    validate_land_time_checks,
+)
+
+# land state variables to determine if land needs to be merged or the end state
+LAND_QUEUED_LABEL = "land-queued"
+LAND_PENDING_LABEL = "land-pending"
+LAND_FAILED_LABEL = "land-failed"
+LAND_SUCCEEDED_LABEL = "land-succeeded"
+
+# land options for checking land checks or all green
+LAND_CHECK_LABEL = 'land-checks'
+ALL_GREEN_LABEL = 'all-green'
+
+PRS_WITH_LABEL_QUERY = (GH_PULL_REQUEST_FRAGMENT
+                        + GH_PR_REVIEWS_FRAGMENT
+                        + GH_CHECKSUITES_FRAGMENT
+                        + GH_COMMIT_AUTHORS_FRAGMENT) + """
+query ($owner: String!, $name: String!, $labels: [String!], $with_labels: Boolean = false) {
+  repository(owner: $owner, name: $name) {
+    pullRequests(first: 10, labels: $labels, states: OPEN){
+        nodes{
+        ...PullRequestFragment
+      }
+    }
+  }
+}
+"""
+
+
+def fetch_land_pending_pr_numbers(org: str, project: str) -> Any:
+    pr_query = gh_graphql(PRS_WITH_LABEL_QUERY,
+                          owner=org,
+                          name=project,
+                          labels=[LAND_PENDING_LABEL, LAND_QUEUED_LABEL],
+                          with_labels=True)
+    return pr_query['data']['repository']['pullRequests']['nodes']
+
+
+def get_github_prs(org: str, project: str, pr_query: Any) -> List[GitHubPR]:
+    prs = []
+    for pr_info in pr_query:
+        prs.append(GitHubPR(org, project, pr_info['number'], pr_info))
+    return prs
+
+
+def is_land_check(labels: List[str]) -> bool:
+    return LAND_CHECK_LABEL in labels
+
+
+def is_all_green(labels: List[str]) -> bool:
+    return ALL_GREEN_LABEL in labels
+
```

Comment:
Probably a good idea to `lintrunner -a .../periodic_merge.py` to make it neater

Reply:
I think I used this but it doesn't make any changes.